pyomo/core/plugins/transform/deactivate_trivial_constraints.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to deactivate trivial constraints."""
+import textwrap
+from math import fabs
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.component_set import ComponentSet
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class TrivialConstraintDeactivator(IsomorphicTransformation):
+    """Deactivates trivial constraints of form constant = constant."""
+
+    alias('core.deactivate_trivial_constraints',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(TrivialConstraintDeactivator, self).__init__()
+        self.tolerance = 1E-13
+
+    def _apply_to(self, instance, tmp=False, ignore_infeasible=False):
+        """Apply the transformation."""
+        if tmp and not hasattr(instance, '_tmp_trivial_deactivated_constrs'):
+            instance._tmp_trivial_deactivated_constrs = ComponentSet()
+
+        for constr in instance.component_data_objects(
+                ctype=Constraint, active=True, descend_into=True):
+            if constr.body.polynomial_degree() == 0:
+                # Check to make sure constraint not violated.
+                if (constr.has_lb() and
+                        value(constr.body) < value(constr.lower)):
```

Comment:
Shouldn't you apply the tolerance to the comparison on line 36 as well?  You check for values close to zero in the next few lines, but why aren't you checking that the value of the body is <= the value of the lower bound plus some tolerance?

Reply:
Good question